### ISSUE TITLE
ldc 1.1.0 beta2/beta3 linux-arm Segmentation fault

### ISSUE THREAD

**Issue by pisetta** [ldc-developers/ldc#1864]
Hi,
I tested that on a Raspberry Pi 3 with ldc2-1.1.0-beta2-linux-arm and ldc2-1.1.0-beta3-linux-arm both segfaulted with this code.

```
import std.stdio;

struct Aggr(alias V) {
    alias Value = V;
}

template templateDecl( Vs... ) {
    void templateDecl() {
        foreach( v; Vs )
            static if ( is( v : Aggr!A, alias A ) )
                write( A );
        writeln();
    }
}

alias templateInst(alias V) = templateDecl!( Aggr!V );

void main() {
    string var = "var";
    templateInst!var();
}
```

I will test if x86 and x86_64 also segfaults soon.

Gianni Pisetta

**Comment by pisetta** [comment]
Hi,
tested the code with linux-x86 and linux-x86_64 and both segfaulted. Attached there is the output of both, if it can be of any help.

Gianni Pisetta
[output-x86.txt](https://github.com/ldc-developers/ldc/files/558388/output-x86.txt)
[output-x86_64.txt](https://github.com/ldc-developers/ldc/files/558387/output-x86_64.txt)

**Comment by LemonBoy** [comment]
The code fails [here](https://github.com/ldc-developers/ldc/blob/master/gen/nested.cpp#L45), `fd` is `<filename>.templateDecl!(Aggr!(var)).templateDecl` whose parent is `null` instead of `main`.
dmd fails with `function D main is a nested function and cannot be accessed from segf.templateDecl!(Aggr!(var)).templateDecl` so it could be considered as a bug in the frontend (the template instantiation is in `main` and I'd expect `main` to be the topmost parent), otherwise a tentative solution would be to change [this line](https://github.com/ldc-developers/ldc/blob/master/gen/nested.cpp#L46) into `!fd || fd->isStatic()`,

**Comment by kinke** [comment]
It may be a front-end bug. Still, I think your proposed change (replace `!fd` with `!vdparent` though) absolutely makes sense. Also note the `assert(vdparent);` a few lines above.

**Comment by LemonBoy** [comment]
Hmm, in this case `vdparent` is not null, the assertion (that today decided it was right about time to fire) fires when checking `fd`:
`ldc2: ../gen/nested.cpp:52: DValue* DtoNestedVariable(Loc&, Type*, VarDeclaration*, bool): Assertion `fd' failed.`

**Comment by kinke** [comment]
Fixed by #1888 (the crash, not the potential DMD front-end issue).
